x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form.tsx

```diff
@@ -113,6 +116,8 @@ export const PipelineForm: React.FunctionComponent<PipelineFormProps> = ({
 
   return (
     <>
+      <UnsavedChangesPrompt hasUnsavedChanges={isFormDirty && !form.isValid} />
```

Comment:
Can you explain why we need to check for both `isFormDirty` and `!form.isValid`?

Reply:
Nice call! Seems I was checking the wrong thing, I shouldve tested deeper than just the field name and description. It seems the processors are tracked outside the form state with a ref value, so we need to account for that as well when we compute the hasUnsavedChanges prop. I've fixed that with https://github.com/elastic/kibana/pull/183699/commits/f66060b3d834807f58158d0ab07392106eea810d

Reply:
> Can you explain why we need to check for both `isFormDirty` and `!form.isValid`?

We need to check if the form is dirty for showing the modal but also need to make sure that the form hasnt been successfully submitted, it also performs a redirect when that happens that can trigger the prompt.